src/Core/Framework/Api/Controller/SalesChannelProxyController.php

```diff
@@ -135,6 +141,56 @@ public function assignCustomer(Request $request, Context $context): Response
         return $response;
     }
 
+    #[Route(path: '/api/_proxy/login-as-customer-token-generate', name: 'api.proxy.login-as-customer-token-generate', methods: ['POST'], defaults: ['_acl' => ['api_proxy_login-as-customer']])]
+    public function loginAsCustomerTokenGenerate(Request $request, Context $context): Response
+    {
+        if (!$request->request->has(self::SALES_CHANNEL_ID)) {
+            throw ApiException::salesChannelIdParameterIsMissing();
+        }
+
+        if (!$request->request->has(self::CUSTOMER_ID)) {
+            throw ApiException::customerIdParameterIsMissing();
+        }
+
+        $customerId = $request->request->getString('customerId');
+        if (!$this->doesCustomerExist($customerId, $context)) {
+            throw ApiException::resourceNotFound('customer', ['id' => $customerId]);
+        }
+
+        $salesChannelId = $request->request->getString(self::SALES_CHANNEL_ID);
+        $salesChannel = $this->fetchSalesChannel($salesChannelId, $context);
+        if ($salesChannel->getDomains() === null || $salesChannel->getDomains()->first() === null) {
+            throw ApiException::invalidSalesChannelId($salesChannelId);
+        }
+
+        $source = $context->getSource();
+        if (!$source instanceof AdminApiSource) {
+            throw new InvalidContextSourceException(AdminApiSource::class, $source::class);
+        }
+
+        $userId = $source->getUserId();
+        if (!$userId) {
+            throw new ExpectedUserHttpException();
+        }
+
+        $token = $this->loginAsCustomerTokenGenerator->generate($salesChannelId, $customerId, $userId);
+
+        $redirectUrlWithToken = sprintf(
+            '%s/%s',
+            $salesChannel->getDomains()->first()->getUrl(),
+            $this->generateUrl('frontend.account.login.customer', [
+                'token' => $token,
+                'customerId' => $customerId,
+                'userId' => $userId,
+            ])
+        );
```

Comment:
The redirect url should be optional. `frontend.account.login.customer` only exists when the Storefront Bundle is installed. 

See https://github.com/shopware/shopware/blob/95bdfc13899c9db4407979e04650e517c1e3ac98/src/Core/System/User/Recovery/UserRecoveryService.php#L80-L85 as an example

Reply:
Changed it to use only the $salesChannel->getDomains()->first()->getUrl() as a fallback URL in https://github.com/shopware/shopware/pull/3727/commits/8972bf7df98128f56a369c5168b5b56fb37e77ae